File: Exercicios/IntroducaoAProgramacaoOrientadaAObjetos/Java/ExHerança/Funcionario.java
package Exercicios.IntroducaoAProgramacaoOrientadaAObjetos.Java.ExHerança;

// Classe abstrata Funcionário implementando a interface Pagamento
public abstract class Funcionario implements Pagamento {
    private String nome;
    private double salario;

    public Funcionario(String nome, double salario) {
        this.nome = nome;
        this.salario = salario;
    }

    public String getNome() {
        return nome;
    }

    public double getSalario() {
        return salario;
    }

    // Método abstrato para calcular o salário anual(deve ser implementado pelas subclasses)
    public abstract double calcularSalarioAnual();

    @Override
    public double calcularPagamentoMensal() {
        return salario; // Retorna o salário mensal padrão
    }
}

File: Exercicios/IntroducaoAProgramacaoOrientadaAObjetos/Python/ExHerança/Funcionario.py
from abc import abstractmethod
from pagamento import Pagamento

# Classe Abstrata Funcionário implementando a interface Pagamento
class Funcionario(Pagamento):
    def __init__(self, nome, salario):
        self.nome = nome
        self.salario = salario

    def get_nome(self):
        return self.nome

    def get_salario(self):
        return self.salario

    # Método abstrato para calcular o salário anual (deve ser implementado pelas subclasses)
    @abstractmethod
    def calcula_salario_anual(self):
        pass

    def calcula_pagamento_mensal(self):
        # Retorna o salário mensal padrão
        return self.salario  
